File: src/main/kotlin/io/energyconsumptionoptimizer/mapservice/application/inbound/FloorPlanService.kt
package io.energyconsumptionoptimizer.mapservice.application.inbound

import io.energyconsumptionoptimizer.mapservice.domain.FloorPlan

/**
 * Service interface for managing floor plan operations.
 *
 */
interface FloorPlanService {
    /**
     * Creates a new floor plan from an SVG representation.
     *
     * @param floorPlanSVG The SVG string representing the house floor plan layout
     * @return The created floor plan entity
     *
     * @throws io.energyconsumptionoptimizer.mapservice.domain.FlorPlanFormatNotValidException
     */
    suspend fun createFloorPlan(floorPlanSVG: String): FloorPlan

    /**
     * Retrieves the current floor plan.
     *
     * @return The floor plan if it exists, null otherwise
     */
    suspend fun getFloorPlan(): FloorPlan?
}

File: src/application/inbound/FloorPlanService.ts
import type { FloorPlan } from "@domain/values/FloorPlan";

/**
 * @inbound
 * Service interface for managing floor plan operations.
 */
export interface FloorPlanService {
  /**
   * Creates a new floor plan from an SVG string.
   * Validates that the content is non-empty and is a valid SVG document.
   *
   * @returns The saved FloorPlan, or an Error (InvalidFloorPlanError) if the SVG is invalid.
   */
  createFloorPlan(floorPlanSVG: string): Promise<FloorPlan | Error>;

  /**
   * Retrieves the current floor plan.
   *
   * @returns The FloorPlan if one exists, or `null` if none has been uploaded yet.
   */
  getFloorPlan(): Promise<FloorPlan | null>;
}
